Source/Core/VideoCommon/XFStructs.cpp

```diff
@@ -300,3 +304,98 @@ void PreprocessIndexedXF(u32 val, int refarray)
   const size_t buf_size = size * sizeof(u32);
   Fifo::PushFifoAuxBuffer(new_data, buf_size);
 }
+
+std::pair<std::string, std::string> GetXFRegInfo(u32 address, u32 value)
+{
+  return std::make_pair(fmt::format("XF reg {:04x} = {:08x}", address, value), "");
+}
+
+std::string GetXFMemName(u32 address)
+{
+  return fmt::format("XF mem address {:04x}", address);
+}
+
+std::string GetXFMemDescription(u32 address, u32 value)
+{
+  return fmt::format("{} = {:08x}", GetXFMemName(address), value);
+}
+
+#define XFMEM_REGISTERS_START 0x1000
+#define XFMEM_REGISTERS_END 0x1058
+
+std::pair<std::string, std::string> GetXFTransferInfo(const u8* data)
```

Comment:
Just a suggestion but since you're appending to a string with `fmt::format`, it would likely be more performant to use `fmt::memory_buffer` with `fmt::format_to`.  Probably negligible but just thought I'd mention :)

Reply:
This code actually comes from PR 9497, but I've changed it to use memory_buffer there (and updated the stub here to match).